sklearn/compose/tests/test_column_transformer.py

```diff
@@ -1007,3 +1093,123 @@ def func(X):
     assert_array_equal(ct.fit(X_df).transform(X_df), X_res_first)
     assert callable(ct.transformers[0][2])
     assert ct.transformers_[0][2] == ['first']
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning:pandas")
+def test_column_transformer_inverse_with_strings():
+    pd = pytest.importorskip('pandas')
+
+    df = pd.DataFrame({
+        'city': ['London', 'London', 'Paris', 'Sallisaw'],
+        'title': ["His Last Bow", "How Watson Learned the Trick",
+                  "A Moveable Feast", "The Grapes of Wrath"],
+        'expert_rating': [5, 3, 4, 5],
+        'user_rating': [4, 5, 4, 3]
+    })
+
+    column_trans = ColumnTransformer(
+        [('city_category', CountVectorizer(analyzer=lambda x: [x]), 'city'),
+         ('title_bow', CountVectorizer(analyzer=lambda x: [x]), 'title')],
+        remainder=MinMaxScaler())
+
+    result = column_trans.fit_transform(df)
+    df_inverse = column_trans.inverse_transform(result)
+
+    pd.util.testing.assert_frame_equal(df, df_inverse)
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning:pandas")
+def test_column_transformer_inverse_with_strings_default_count_vectorizer():
+    pd = pytest.importorskip('pandas')
+
+    df = pd.DataFrame({
+        'city': ['London', 'London', 'Paris', 'Sallisaw'],
+        'title': ["His Last Bow", "How Watson Learned the Trick",
+                  "A Moveable Feast", "The Grapes of Wrath"],
+        'expert_rating': [5, 3, 4, 5],
+        'user_rating': [4, 5, 4, 3]
+    })
+
+    column_trans = ColumnTransformer(
+        [('city_category', CountVectorizer(), 'city'),
+         ('title_bow', CountVectorizer(), 'title')],
+        remainder=MinMaxScaler())
+
+    result = column_trans.fit_transform(df)
+    df_inverse = column_trans.inverse_transform(result)
+
+    name_to_cv = {k: v for k, v, _ in column_trans.transformers_}
+    title_cv = name_to_cv['title_bow']
+    title_result = title_cv.transform(df['title'])
+    title_inverse = title_cv.inverse_transform(title_result)
+
+    city_cv = name_to_cv['city_category']
+    city_result = city_cv.transform(df['city'])
+    city_inverse = city_cv.inverse_transform(city_result)
+
+    expected_inverse = df.copy()
+    expected_inverse['title'] = np.array(title_inverse)
+    expected_inverse['city'] = np.array(city_inverse)
+
+    pd.util.testing.assert_frame_equal(df_inverse, expected_inverse)
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning:pandas")
+def test_column_transformer_inverse_with_strings_passthrough():
```

Comment:
maybe this should be combined with `test_column_transformer_inverse_with_strings` since, again, the only difference is the `column_trans` parameters.